build/configs/common_download.sh

```diff
@@ -0,0 +1,390 @@
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+# common_download.sh
+
+CURDIR=$(readlink -f "$0")
+CUR_PATH=$(dirname "$CURDIR")
+BIN_PATH=${CUR_PATH}/../output/bin
+TOP_PATH=${CUR_PATH}/../..
+OS_PATH=${TOP_PATH}/os
+
+CONFIG=${OS_PATH}/.config
+source ${CONFIG}
+SMARTFS_BIN_PATH=${BIN_PATH}/${CONFIG_ARCH_BOARD}_smartfs.bin
+
+source ${CUR_PATH}/${CONFIG_ARCH_BOARD}/${CONFIG_ARCH_BOARD}_download.sh
+source ${CUR_PATH}/${CONFIG_ARCH_BOARD}/binary_name_map.txt
+
+## For manual port selection, set $3 as the portname, eg. make download ALL x ttyUSB2
+
+if [ -n "$3" ]; then
+	PORT="$3"
+else
+	PORT=${DEFAULT_PORT}
+fi
+
+WARNING="\n Port $PORT is selected\n\n
+	############################################\n
+	WARNINGS:\n
+	1. Make sure the board is in DOWNLOAD MODE.\n
+	2. Make sure NO other application like putty,\n
+	is occupying $PORT.\n
+	############################################\n"
+
+
+TTYDEV="/dev/${PORT}"
+
+##Utility function for sanity check##
+function sanity_check()
+{
+	if [ ! -f ${CONFIG} ];then
+		echo "No .config file"
+		exit 1
+	fi
+	
+	CHECK_BOARD=$(echo "CONFIG_ARCH_BOARD_${BOARD}")
+	if [[ "${!CHECK_BOARD}" != "y" ]];then
+		echo "Target is NOT ${BOARD}"
+		exit 1
+	fi
+
+	if fuser -s ${TTYDEV};then
+		echo "${TTYDEV} is used by another process, can't proceed"
+		exit 1
+	fi
+}
+
+##Utility function to match partition name to binary name##
+function get_executable_name()
+{
+	case $1 in
+		bl1) echo "${BL1}.bin";;
+		bl2) echo "${BL2}.bin";;
+		kernel|ota) echo "${KERNEL}.bin";;
+		micom) echo "micom";;
+		wifi) echo "wifi";;
+		zoneinfo) echo "zoneinfo.img";;
+		bootparam) echo "${BOOTPARAM}.bin";;
+		userfs) echo "${CONFIG_ARCH_BOARD}_smartfs.bin";;
+		*) echo "No Binary Match"
+		exit 1
+	esac
+}
+
+flash_ota=false
+##Utility function to get partition index ##
+function get_partition_index()
+{
+	for idx in ${!parts[@]}; do
+		if [[ ${parts[$idx],,} == ${1,,} ]]; then
+			if [[ $flash_ota == true ]]; then
+				## In this case, try to find 2nd kernel partition for OTA
+				flash_ota=false
+				continue;
+			fi
+			echo $idx
+			exit 1
+		fi
+	done
+	echo -1
+}
+
+##Help utility##
+function dwld_help()
+{
+        cat <<EOF
+	HELP:
+		make download ALL or [PARTITION(S)]
+		make download ERASE ALL or erase [PARTITION(S)]
+
+	PARTITION(S):
+		[${uniq_parts[@]}]  NOTE:case sensitive
+
+	For examples:
+		make download ALL
+		make download ERASE ALL
+		make download kernel
+		make download erase kernel
+		make download ota
+		make download erase ota
+		make download erase ss
+		make download smartfs
```

Comment:
I think this is wrong since we dont have a partition named as smartfs. It is always userfs. So let us remove this line.